docs/regression_test_api.rst

```diff
@@ -87,6 +87,40 @@ For instance, continuing with the example above, one could override the :func:`_
        This only has an effect if used with ``inherit_params=True``.
 
 
+.. py:function:: reframe.core.pipeline.RegressionTest.var(name, *types, value=None, field=None)
+
+   Inserts a new regression test variable.
+   If the ``value`` argument is not provided, the variable is considered *declared* but not *defined*.
+   Thus, a variable may not be declared more than once. However, it is possible to alter a variable's value after it was declared by using the :func:`set_var` and :func:`require_var` directives.
+   Note that a variable must be defined before is referenced in the regression test. Otherwise, an :py:exc:`AttributeError` will be raised.
+
+   :param name: the variable name.
+   :param types: the supported types for the variable.
+   :param value: the default value assigned to the variable.
+   :param field: the field validator to be used for this variable.
+        If no field argument is provided, it defaults to
+        :class:`reframe.core.fields.TypedField`.
+        Note that the field validator provided by this argument must derive from
+        :class:`reframe.core.fields.Field`.
+
+
+.. py:function:: reframe.core.pipeline.RegressionTest.set_var(name, value)
+
+   Assign a value to a regression test variable. The variable must have been defined in a parent class using the :func:`var` directive.
+
+   :param name: the variable name.
+   :param value: the value assigned to the variable.
+
+
+.. py:function:: reframe.core.pipeline.RegressionTest.require_var(name)
+
+   Undefine a regression test variable. The variable must have been defined in a parent class using the :func:`var` directive.
```

Comment:
```suggestion
   Make variable `name` required for this class and any descendants.
```

Then you can talk about the "undefining" etc., which I think is an implementation detail.